schema/compose-spec.json

```diff
@@ -589,6 +590,23 @@
         "patternProperties": {"^x-": {}}
       }
     },
+    
+    "devices": {
+      "id": "#/definitions/devices",
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+            "capabilities": {"$ref": "#/definitions/list_of_strings"},
+            "count": {"type": "integer"},
```

Comment:
[count can also be __all__](https://github.com/compose-spec/compose-spec/blob/master/deploy.md#count) to indicate that all devices should be allocated. Could we adjust the schema to allow for that?